### PR TITLE
Issue 939: adjust hysds payload for proper deduping

### PR DESCRIPTION
**Pull request by chrisjrd**
## Purpose
- fix dedupe behavior when set coverage improves
## Proposed Changes
- [CHANGE] payload to change when set coverage improves

## Issues
Refs #939 
## Testing
- tested locally

### CODE REVIEW

File: data_subscriber/rtc/rtc_download_job_submitter.py [hunk 1]
```diff
@@ -33,6 +36,11 @@ def submit_rtc_download_job_submissions_tasks(batch_id_to_products_map, args, se
                 "metadata": {
                     "batch_id": mgrs_set_id_acquisition_ts_cycle_index,
                     "mgrs_set_id": mgrs_set_id,
+                    # for payload hash dedupe, include granule ID list (changes with improved coverage)
+                    "granule_ids": sorted({
```
Comment (riverma) [inline data_subscriber/rtc/rtc_download_job_submitter.py:40]: Could you explain how the code that interprets `granule_ids` and incorporates it into the logic that prevents the dedupe works? This is simply setting the metadata. Is the logic within HySDS?

Comment (chrisjrd) [reply data_subscriber/rtc/rtc_download_job_submitter.py:40]: HySDS logic. It calculates a payload hash. this change purposely makes makes the payload change as more or different products are added to a set.

**Comment by riverma** [comment]
@chrisjrd @hhlee445 - can we add this as an integration test case in https://github.com/nasa/opera-sds-pcm/blob/develop/tests/integration/test_integration.py or equivalent? In general - it's good software engineering practice to ensure bugs that are found are added to automated test cases to prevent regression in the future.

**Comment by philipjyoon** [comment]
@chrisjrd have you verified that the corresponding SCIFLO triggering is also not deduped? Or does that also need to be changed up a bit?

**Comment by hhlee445** [comment]
Verified that the second job didn't get deduped and generated more DSWx-S1 products by later job

1. 'MS_147_4$168', 'e5540925-ff88-4c0c-8182-b6ea374dd1df' was triggered with

python3 ~/mozart/ops/opera-pcm/data_subscriber/daac_data_subscriber.py query -c OPERA_L2_RTC-S1_V1   --job-queue=opera-job_worker-rtc_data_download  --chunk-size 1 --processing-mode=forward  -s 2024-07-19T00:15:19Z -e 2024-07-19T01:15:19Z

MS_147_4 - 27 RTCs were used

2. 'MS_147_4$168', '29a96e5b-ffc7-409b-9e9c-1c878a8494b7' was triggered with

python3 ~/mozart/ops/opera-pcm/data_subscriber/daac_data_subscriber.py query -c OPERA_L2_RTC-S1_V1   --job-queue=opera-job_worker-rtc_data_download  --chunk-size 1 --processing-mode=forward  -s 2024-07-19T07:15:19Z -e 2024-07-19T08:15:19Z

MS_147_4 - 41 RTCs were used



Screenshot 2024-08-07 at 2 30 55 PM

**Review by riverma — APPROVED** [review]

**Review by hhlee445 — APPROVED** [review]

**Pull request merged**